windows/utils/win_get_eflow_vm_state.yml

```diff
@@ -0,0 +1,27 @@
+# Copyright 2022 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Get EFLOW VM power state info in Windows guest OS
+#
+- name: "Initialize the power state of EFLOW virtual machine"
+  ansible.builtin.set_fact:
+    win_eflow_vm_power_state: ''
+
+- name: "Get EFLOW virtual machine info"
+  include_tasks: win_execute_cmd.yml
+  vars:
+    win_powershell_cmd: "Get-EflowVM | Format-List"
+
+- name: "Get EFLOW virtual machine power state"
+  block:
+    - name: "Set fact of EFLOW virtual machine power state"
+      ansible.builtin.set_fact:
+        win_eflow_vm_power_state: "{{ item.split(':')[1].strip() }}"
+      when: item.split(':')[0].strip() == 'VmPowerState'
+      with_items: "{{ win_powershell_cmd_output.stdout_lines | select }}"
```

Comment:
I think you can use 
```
win_powershell_cmd_output.stdout_lines | select('match', '\\s*VmPowerState\\s*:') 
```
to find the line directly.

Reply:
This will define 2 variables, but seems more clear. Thanks.